CONTRIBUTING.md

```diff
@@ -0,0 +1,79 @@
+# Contributing to `PIXL`.
+
+👍🎉 First off, thanks for taking the time to contribute! 🎉👍
+
+When contributing to this repository, please first discuss the change you wish to make via issue, instant message in clara-agx chatbox, or any other method with the owners of this repository before making a change. 
+Please note we have a [code of conduct](CODE_OF_CONDUCT.md), please follow it in all your interactions with the project.
+
+## :octocat: Setting up project locally. 
+1. Generate your SSH keys as suggested [here](https://docs.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+2. Clone the repository by typing (or copying) the following lines in a terminal
+```
+git clone git@github.com:UCLH-Foundry/PIXL.git
+```
+
+## New issues
+* Open an issue (bug report, feature request, or something is not working): https://github.com/UCLH-Foundry/PIXL/issues/new/choose 
+* Workflow for issue management 
+```mermaid
+  flowchart TD;
+      Z[Bug Reported] -->A[...];  
+      A[Bug resolution] -->B(Testing OK?);
+      B--Yes-->C[Prepare commit];
+      B--No-->D[Reopen issue];
+      D----> A[Bug resolution];
+      C ----> E[Request Review];
+      E ----> F[Commit changes];
+      F --> H[Merge PR and close issue]
+```  
+
+## Committing and pushing changes 
+1. Clone this repo 
+```
+git clone git@github.com:UCLH-Foundry/PIXL.git
+``` 
+2. Create new branch using issue number
+```
+git checkout -b ISSUENUMBER-branch-name 
+```
+3. Commit changes and push to your branch
+```
+git add .
+git commit -m 'short message (#ISSUENUMBER)'
+git push origin ISSUENUMBER-branch-name
+```
+4. Submit a Pull Request against the `main` branch. 
+
+## Pull Request (PR) and merge to `main` branch
+1. Select branch that contain your commits.
+2. Click `Compare and pull request` and create PR for the associated branch.
+3. Type a title and description of your PR and create PR
+4. Please keep your PR in sync with the base branch.
+```
+git checkout main
+git pull origin main
+git checkout FEATURE_BRANCH
+git rebase main
+git push --force origin FEATURE_BRANCH
+```
+4.1 In case you are in a different `MY_FEATURE_BRANCH` branch, follow:
+```
+git checkout FEATURE_BRANCH
+git pull origin FEATURE_BRANCH
+git checkout MY_FEATURE_BRANCH 
+git rebase FEATURE_BRANCH
+git push --force origin MY_FEATURE_BRANCH
```

Comment:
for these we use squash merge, so we can just pull in the changes and merge the main branch into it (as the PR will only create a single commit in main when merged). To me at least that's easier to talk people through rather than rebasing if its a long lived feature branch

Reply:
resolved here 16c1f6dea29535000d581234f68de7af72dcbcad